### ISSUE TITLE
vim: Add sneak motion

### ISSUE THREAD

**Issue by mrkajetanp** [zed-industries/zed#13858]
### Check for existing issues

- [X] Completed

### Describe the feature

The vim-sneak plugin provides a nice 'sneak' motion which jumps to any location specified by two characters. It shouldn't be very complicated to implement (hopefully) and would be a nice addition - I might try to give it a go myself when I have some spare time but I thought it might be worth opening an issue for it regardless.

https://github.com/justinmk/vim-sneak/tree/master

**Comment by ConradIrwin** [comment]
@mrkajetanp definitely. If you want to pair on this: https://calendly.com/conradirwin/pairing.

**Comment by mrkajetanp** [comment]
I was curious so I just went ahead and implemented it, the PR is linked above :)
